src/context.tsx

```diff
@@ -13,22 +13,44 @@ interface PortalMap {
   [name: string]: React.ReactNode;
 }
 
+type PortalAction = {
+  type: 'portals/SET';
+  name: string;
+  children: React.ReactNode;
+};
```

Comment:
not a blocker: I wonder if it's worth having a `portals/REMOVE` action that actually `omit`s the key so that we're not rendering empty exit portals for old `name`s. 🤔

Reply:
As in explicitly setting the key to `undefined` and then having `<ExitPortal />` explicitly render `null` if it can't find children in the map?

Reply:
I'm not opposed to doing that but I would consider that an optimization that is not necessary for this PR.